flink-libraries/flink-table/src/main/scala/org/apache/flink/table/runtime/functions/ScalarFunctions.scala

```diff
@@ -196,9 +196,35 @@ object ScalarFunctions {
     new String(data)
   }
 
+  /**
+    * Returns the numeric value of the leftmost character of the string str.
+    * Returns 0 if str is the empty string. Returns NULL if str is NULL.
+    */
+  def ascii(str: String): String = {
+    if (str == null) {
+      null
+    } else if (str.equals("")) {
+      ""
+    } else {
+      str.charAt(0).toByte.toString
+    }
+  }
+
   /**
     * Returns the base string decoded with base64.
     */
   def fromBase64(str: String): String = new String(Base64.decodeBase64(str))
 
+  /**
+    * Returns string contains a character which converts from a ASCII integer.
```

Comment:
Returns the character corresponding to the input ASCII integer. If the ASCII less then 0 or greater than 255, returns null.